v2/managedsettling.go

```diff
@@ -27,9 +27,11 @@ func (m *ManagedSettler) Handle(ctx context.Context, settler MessageSettler, mes
 	}
 	if err := settler.CompleteMessage(ctx, message, nil); err != nil {
 		log(ctx, err)
+		m.options.OnFailure(ctx, message, err)
```

Comment:
should we have the same OnFailure func to run after deadlettering AND after completing the message? I feel like a failure to call settler.CompleteMessage is a different kind of error/a sign of an internal issue as opposed to whatever may arise from failing to handle the message within the user-provided `ManagedSettlingFunc`

Reply:
I think that's enough at the moment yes. the user can still infer the deadletter because he controls the canretry func, and can inspect the message. in our usecase in fleet I don't feel like we need more at this point